### PR TITLE
Feat/route list calendar

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Remove previous calendar ui for new one
- Make changes with default css react range styles
- Add buttons for apply and cancel, separate logic to hook

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-131)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

photo_2024-12-10_05-26-26

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/components/CalendarRange/style.css [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.rdrDayHovered .rdrDayStartPreview,
+.rdrDayHovered .rdrDayEndPreview {
+  background-color: rgba(115, 103, 240, 0.16);
```
Comment (Parle648) [inline src/components/CalendarRange/style.css]: 1.3.2

File: src/components/CalendarRange/style.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+.rdrStartEdge,
+.rdrEndEdge {
+  background-color: rgba(115, 103, 240, 1);
+  color: rgba(255, 255, 255, 1) !important;
```
Comment (Parle648) [inline src/components/CalendarRange/style.css]: 1.3.2

File: src/components/CalendarRange/style.css (hunk 1 above)
Comment (oyakovytskyi) [reply src/components/CalendarRange/style.css]: create constants

File: src/components/CalendarRange/style.css (hunk 2 above)
Comment (oyakovytskyi) [reply src/components/CalendarRange/style.css]: done

**Review by Parle648 — APPROVED** [review]

**Review by AlexeyBezrebriy — APPROVED** [review]

**Pull request merged**